File: web/src/lib/types/notification.ts
export type NotificationType =
	| 'subscription'
	| 'donation'
	| 'bits'
	| 'giveaway_winner'
	| 'generic';

export type NotificationRecord = {
	id: number;
	type: NotificationType;
	platform?: string;
	username?: string;
	amount?: number;
	message?: string;
	metadata?: Record<string, string>;
	created_at: string;
};

export type CreateNotificationPayload = {
	type: NotificationType;
	platform?: string;
	username?: string;
	amount?: number;
	message?: string;
	metadata?: Record<string, string>;
};

File: internal/domain/notification.go
package domain

import "time"

type NotificationType string

const (
	NotificationSubscription   NotificationType = "subscription"
	NotificationDonation       NotificationType = "donation"
	NotificationBits           NotificationType = "bits"
	NotificationGiveawayWinner NotificationType = "giveaway_winner"
	NotificationGeneric        NotificationType = "generic"
)

type Notification struct {
	ID        int64
	Type      NotificationType
	Platform  Platform
	Username  string
	Amount    float64
	Message   string
	Metadata  map[string]string
	CreatedAt time.Time
}
